test/hotspot/jtreg/testlibrary_tests/ir_framework/examples/SetupExample.java

```diff
@@ -0,0 +1,158 @@
+ * @see Test
+ */
+public class SetupExample {
+    private static final Random RANDOM = Utils.getRandomInstance();
+
+    int iFld1, iFld2;
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    // ----------------- Random but Linked --------------
+    @Setup
+    static Object[] setupLinkedII() {
+        int r = RANDOM.nextInt();
+        return new Object[]{ r, r + 42};
+    }
+
+    @Test
+    @Arguments(setup = "setupLinkedII")
+    static int testSetupLinkedII(int a, int b) {
+        return b - a;
+    }
+
+    @Check(test = "testSetupLinkedII")
+    static void checkSetupLinkedII(int res) {
+        if (res != 42) { throw new RuntimeException("wrong result " + res); }
+    }
+
+    // ----------------- Random Arrays --------------
+    static int[] generateI(int len) {
+        int[] a = new int[len];
+        for (int i = 0; i < len; i++) {
+            a[i] = RANDOM.nextInt();
+        }
+        return a;
+    }
+
+    @Setup
+    static Object[] setupRandomArrayII() {
+        // Random length, so that AutoVectorization pre/main/post and drain loops are tested
+        int len = RANDOM.nextInt(20_000);
+        int[] a = generateI(len);
+        int[] b = generateI(len);
+        return new Object[] { a, b };
+    }
+
+    @Test
+    @Arguments(setup = "setupRandomArrayII")
+    static Object[] testAdd(int[] a, int[] b) {
+        int[] c = new int[a.length];
+        for (int i = 0; i < a.length; i++) {
+            c[i] = a[i] + b[i];
+        }
+        return new Object[] { a, b, c };
+    }
+
+    @Check(test = "testAdd")
+    static void checkAdd(Object[] res) {
+        int[] a = (int[])res[0];
+        int[] b = (int[])res[1];
+        int[] c = (int[])res[2];
+        for (int i = 0; i < a.length; i++) {
+            if (c[i] != a[i] + b[i]) {
+                throw new RuntimeException("wrong values: " + a[i] + " " + b[i] + " " + c[i]);
+            }
+        }
+    }
+
+    // ----------------- Setup Fields ---------------
+    @Setup
+    void setupFields() {
+        int r = RANDOM.nextInt();
+        iFld1 = r;
+        iFld2 = r + 42;
+    }
+
+    @Test
+    @Arguments(setup = "setupFields")
+    int testSetupFields() {
+        return iFld2 - iFld1;
+    }
+
+    @Check(test = "testSetupFields")
+    static void checkSetupFields(int res) {
+        if (res != 42) { throw new RuntimeException("wrong result " + res); }
+    }
+
+    // ----------------- Deterministic Values -------
+    @Setup
+    Object[] setupDeterministic(SetupInfo info) {
+        // This value increments with every invocation of the setup method: 0, 1, 2, ...
+        int cnt = info.invocationCounter();
+
+        // Return true with low frequency. If we did this randomly, we can get unlucky
+        // and never return true. So doing it deterministically can be helpful when we
+        // want "low frequency" but a guaranteed "true" at some point.
+        return new Object[]{ (long)(cnt % 1_000) };
```

Comment:
Any specific reason that this should be a `long` and not an `int` (i.e. also change `testLowProbabilityBranchDeterministic()` to take an int?